docs/v0.6/en/user-guide/client-libraries/go.md

```diff
@@ -36,153 +37,256 @@ import (
-seriesHost2.SetTimestamp(time.Now())
-
-metric := greptime.Metric{}
-metric.AddSeries(seriesHost1, seriesHost2)
+ts := time.Now()
+_ = cpuMetric.AddRow("127.0.0.1", ts, 0.1, 0.12)
+// insert a row data
+resp, _ := cli.Write(context.Background(), cpuMetric)
+
+// update the row data
+// The same tag `127.0.0.1`
+// The same time index `ts`
+// The new value: cpu_user = `0.80`, cpu_sys = `0.11`
+_ = cpuMetric.AddRow("127.0.0.1", ts, 0.80, 0.11)
+// overwrite the existing data
+resp, _ = cli.Write(context.Background(), cpuMetric)
 ```
 
 %}
 
 
-{template save-rows%
+{template orm-style-object%
 
 ```go
-seriesHost1 := greptime.Series{}
-seriesHost1.AddStringTag("host", "host1")
-seriesHost1.AddFloatField("cpu", 0.90)
-seriesHost1.SetTimestamp(time.Now())
+type CpuMetric struct {
+    Host            string    `greptime:"tag;column:host;type:string"`
+    CpuUser         float64   `greptime:"field;column:cpu_user;type:float64"`
+    CpuSys          float64   `greptime:"field;column:cpu_sys;type:float64"`
+    Ts              time.Time `greptime:"timestamp;column:ts;type:timestamp;precision:millisecond"`
+}
 
-seriesHost2 := greptime.Series{}
-seriesHost2.AddStringTag("host", "host2")
-seriesHost2.AddFloatField("cpu", 0.70)
-seriesHost2.SetTimestamp(time.Now())
+func (CpuMetric) TableName() string {
+	return "cpu_metric"
+}
 
-metric := greptime.Metric{}
-metric.AddSeries(seriesHost1, seriesHost2)
+cpuMetrics := []CpuMetric{
+    {
+        Host:        "127.0.0.1",
+        CpuUser:     0.10,
+        CpuSys:      0.12,
+        Ts:          time.Now(),
+    }
+}
 
-monitorReq := greptime.InsertRequest{}
-monitorReq.WithTable("monitor").WithMetric(metric)
 
-insertsRequest := greptime.InsertsRequest{}
-insertsRequest.Append(monitorReq)
+type MemMetric struct {
+    Host        string    `greptime:"tag;column:host;type:string"`
+	Memory      float64   `greptime:"field;column:mem_usage;type:float64"`
+	Ts          time.Time `greptime:"timestamp;column:ts;type:timestamp;precision:millisecond"`
+}
 
-res, err := client.Insert(context.Background(), insertsRequest)
-if err != nil {
-    fmt.Printf("fail to insert, err: %+v\n", err)
-    // error handling
-    // ...
+func (MemoryMetric) TableName() string {
+	return "mem_metric"
+}
+
+memMetrics := []MemMetric{
+    {
+        Host:        "127.0.0.1",
+        Memory:      112,
+        Ts:          time.Now(),
+    }
 }
-fmt.Printf("AffectedRows: %d\n", res.GetAffectedRows().Value)
+```
+%}
+
+{template orm-style-insert-data%
+
+```go
+resp, err := cli.Create(context.Background(), cpuMetrics, memMetrics)
+log.Printf("affected rows: %d\n", resp.GetAffectedRows().GetValue())
 ```
 
 %}
 
-{template update-rows%
+{template orm-style-streaming-insert%
+
+To insert data using the streaming API, you must first create a stream client.
+
+```go
+cfg := config.New("<host>").
+    WithAuth("<username>", "<password>").
+    WithDatabase(database)
+streamClient, err := client.NewStreamClient(cfg)
```

Comment:
too many config and client initiation. Is it better to split a code block to show how to construct a client, then all the following code blocks use the `client` variable?